src/js/components/datepicker/datepicker_state.js

```diff
@@ -30,6 +30,26 @@ export default Backbone.Model.extend({
 
     this.set('currentMonth', currentMonth.startOf('month'));
   },
+  setSelectedDate(date) {
+    if (date) {
+      date = dayjs(date).startOf('day');
+    }
+
+    this.set({
```

Comment:
So let's make this more like the month one and update both..
I think it'll need to be

```js
if (!date) {
  this.set({
    selectedDate: null,
    currentMonth: dayjs().startOf('month'),
  });
  return;
}

date = dayjs(date);

this.set({
  selectedDate: date.startOf('day'),
  currentMonth: date.startOf('month'),
  selectedMonth: null,
});
```

Then I think we can remove the two change events above and rely on this new api.
but it means we should add both
`setSelectedMonth` and `setSelectedDate` as the first things in the initialize and remove the current `set` so that any strings are converted to dayjs.  And that should remove the need to call `setCurrentMonth`